File: frontend/src/types/agent.ts
export interface Agent {
  id: number
  project_id: number
  name: string
  role_name: string
  system_prompt: string | null
  background_identity: string | null
  background_experience: string | null
  domain_knowledge: string | null
  responsibility: string | null
  constraints: string | null
  model_id: number | null
  model_name?: string
  allow_tool_use: boolean
  skill_ids: number[]
  tool_ids: number[]
  created_at: string | null
  updated_at: string | null
}

export interface AgentCreateRequest {
  name: string
  project_id: number
  role_name?: string
  system_prompt?: string
  background_identity?: string
  background_experience?: string
  domain_knowledge?: string
  responsibility?: string
  constraints?: string
  model_id?: number
  allow_tool_use?: boolean
  skill_ids?: number[]
  tool_ids?: number[]
}

export interface AgentUpdateRequest {
  name?: string
  role_name?: string
  system_prompt?: string
  background_identity?: string
  background_experience?: string
  domain_knowledge?: string
  responsibility?: string
  constraints?: string
  model_id?: number
  allow_tool_use?: boolean
}

File: backend/app/schemas/agent.py
from datetime import datetime

from pydantic import BaseModel


class AgentCreateRequest(BaseModel):
    name: str
    project_id: int
    role_name: str = "assistant"
    system_prompt: str | None = None
    background_identity: str | None = None
    background_experience: str | None = None
    domain_knowledge: str | None = None
    responsibility: str | None = None
    constraints: str | None = None
    model_id: int | None = None
    allow_tool_use: bool = False
    skill_ids: list[int] | None = None
    tool_ids: list[int] | None = None
    role_template_id: int | None = None
    description: str | None = None


class AgentUpdateRequest(BaseModel):
    name: str | None = None
    role_name: str | None = None
    system_prompt: str | None = None
    background_identity: str | None = None
    background_experience: str | None = None
    domain_knowledge: str | None = None
    responsibility: str | None = None
    constraints: str | None = None
    model_id: int | None = None
    allow_tool_use: bool | None = None
    role_template_id: int | None = None


class AgentSkillBindRequest(BaseModel):
    skill_ids: list[int]


class AgentToolBindRequest(BaseModel):
    tool_ids: list[int]


class AgentResponse(BaseModel):
    id: int
    project_id: int
    name: str
    role_name: str
    system_prompt: str | None = None
    background_identity: str | None = None
    background_experience: str | None = None
    domain_knowledge: str | None = None
    responsibility: str | None = None
    constraints: str | None = None
    model_id: int | None = None
    allow_tool_use: bool = False
    skill_ids: list[int] = []
    tool_ids: list[int] = []
    role_template_id: int | None = None
    description: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None

    model_config = {"from_attributes": True}


class AgentListResponse(BaseModel):
    total: int
    items: list[AgentResponse]
